src/rgw/rgw_admin.cc

```diff
@@ -2509,7 +2509,8 @@ static int bucket_source_sync_status(const DoutPrefixProvider *dpp, rgw::sal::Ra
   out << indented{width, "source zone"} << source.id << " (" << source.name << ")" << std::endl;
 
   // syncing from this zone?
-  if (!zone.syncs_from(source.name)) {
+  if (!zone.syncs_from(source.name) ||
+     (!static_cast<rgw::sal::RadosStore*>(driver)->svc()->zone->zone_syncs_from(static_cast<rgw::sal::RadosStore*>(driver)->svc()->zone->get_zone(), source))) {
```

Comment:
```c++
if (!driver->svc()->zone->zone_syncs_from(zone, source)) {
```
* `driver` is already a `RadosStore*` so we don't need the casts
* `RGWSI_Zone::zone_syncs_from()` already checks `target_zone.syncs_from(source_zone)`
* `driver->svc()->zone->get_zone()` is just `zone` here

Reply:
thanks! addressed it.